activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java

```diff
@@ -1630,6 +1635,24 @@ public void setDeleteAllMessagesOnStartup(boolean deletePersistentMessagesOnStar
         this.deleteAllMessagesOnStartup = deletePersistentMessagesOnStartup;
     }
 
+    public void deleteAllScheduledMessages() throws Exception {
+        if (getJobSchedulerStore() != null) {
+            getJobSchedulerStore().getJobScheduler(SchedulerBroker.INTERNAL_JOB_SCHEDULER_NAME).removeAllJobs();
+        }
+    }
+
+    public boolean isDeleteAllScheduledMessagesOnStartup() {
+        return deleteAllScheduledMessagesOnStartup;
+    }
+
+    /**
+     * Sets whether or not all scheduled messages are deleted on startup
+     * @org.apache.xbean.Property propertyEditor="org.apache.activemq.util.BooleanEditor"
+     */
+    public void setDeleteAllScheduledMessagesOnStartup(boolean deleteAllScheduledMessagesOnStartup) {
+        this.deleteAllScheduledMessagesOnStartup = deleteAllScheduledMessagesOnStartup;
+    }
+
```

Comment:
The `addInterceptors` method calls `sb.setMaxRepeatAllowed(maxSchedulerRepeatAllowed);`.
Would it be possible to also expose a `setDeleteAllScheduledMessagesOnStartup` method on `SchedulerBroker`? Then move the delete functionality into  `SchedulerBroker` and invoke it on `start()` if `deleteAllScheduledMessagesOnStartup` is true.

Reply:
That's a good idea actually. I took a brief look before and my original concern is should the scheduler be aware of messages or it should be generic to job. But taking a second look it seems like it has message related operation there, so I will use this approach. And then I will raise a draft PR in upstream asking for maintainer opinion.